GliaWidgets/Sources/AlertManager/AlertManager.swift

```diff
@@ -0,0 +1,212 @@
+                viewController.view.bottomAnchor.constraint(equalTo: alertViewController.view.bottomAnchor),
+                viewController.view.leadingAnchor.constraint(equalTo: alertViewController.view.leadingAnchor),
+                viewController.view.trailingAnchor.constraint(equalTo: alertViewController.view.trailingAnchor)
+            ])
+        default:
+            let completion: () -> Void = {
+                viewController.present(
+                    alertViewController,
+                    animated: true
+                )
+            }
+            guard let presented = viewController.presentedViewController as? Replaceable else {
+                completion()
+                return
+            }
+            presented.dismiss(
+                animated: true,
+                completion: completion
+            )
+        }
+    }
+
+    func presentGlobally(
+        type: AlertType,
+        input: AlertInputType
+    ) {
+        let alertViewController = createAlertViewController(type: type)
+        alertViewController.onDismissed = { [weak self] in
+            self?.cleanup()
+        }
+        guard isAlertReplacable(offer: alertViewController) else { return }
+        self.alertViewController = alertViewController
+        self.currentAlert = input
+        createAlertWindow()
+        guard let alertWindow = alertWindow else { return }
+        alertWindow.isHidden = false
+        alertWindow.makeKeyAndVisible()
+        alertWindow.rootViewController?.present(
+            alertViewController,
+            animated: true,
+            completion: nil
+        )
+    }
+
+    func createAlertViewController(type: AlertType) -> AlertViewController {
+        .init(type: type, viewFactory: viewFactory)
+    }
+
+    func createSystemAlert(
+        conf: SettingsAlertConfiguration,
+        cancelled: (() -> Void)? = nil
+    ) -> UIAlertController {
+        let alert = UIAlertController(
+            title: conf.title,
+            message: conf.message,
+            preferredStyle: .alert
+        )
+        let cancel = UIAlertAction(
+            title: conf.cancelTitle,
+            style: .cancel,
+            handler: { _ in
+                cancelled?()
+            }
+        )
+        let settings = UIAlertAction(
+            title: conf.settingsTitle,
+            style: .default,
+            handler: { _ in
+                guard let settingsURL = URL(string: UIApplication.openSettingsURLString) else { return }
+                UIApplication.shared.open(settingsURL)
+            }
+        )
+        alert.addAction(cancel)
+        alert.addAction(settings)
+
+        return alert
+    }
+
+    func isAlertReplacable(offer: Replaceable) -> Bool {
+        guard let alertViewController else { return true }
+        switch alertViewController.isReplaceable(with: offer) {
+        case true:
+            return true
+        case false:
+            return false
+        }
```

Comment:
Does it make sense?
```suggestion
        alertViewController.isReplaceable(with: offer) ?? true
```

Reply:
fixed [c08f255](https://github.com/salemove/ios-sdk-widgets/pull/968/commits/c08f255d539b691b212f88c3695b2f272a154aad)